src/wallet/spend.cpp

```diff
@@ -767,10 +783,32 @@ static bool CreateTransactionInternal(
 
     // Get available coins
     std::vector<COutput> vAvailableCoins;
-    AvailableCoins(wallet, vAvailableCoins, &coin_control, 1, MAX_MONEY, MAX_MONEY, 0);
 
-    // Choose coins to use
-    std::optional<SelectionResult> result = SelectCoins(wallet, vAvailableCoins, /* nTargetValue */ selection_target, coin_control, coin_selection_params);
+    // First, do coin selection on a filtered subset of coins, filtered by OutputType
+    // Attempt with older output types first (LEGACY > P2SH > BECH32 > BECH32M)
+    std::optional<SelectionResult> result = [&] {
+        // This takes advantange of the fact that OUTPUT_TYPES is already sorted by age (LEGACY, P2SH, BECH32, BECH32M)
```

Comment:
This assumes there is a strict preference for newer UTXO types. Rather, it should try to prefer using UTXOs that are not the user's preferred address type(s).

Reply:
sorry, forgot to respond to this! yes, rather than rely on a forced ordering of old to new, I'm implementing @Xekyo 's suggestion of running coin selection on each output group and then picking the solution with the best waste metric. If no solution is found, then run coin selection over all coins (current behavior).

Reply:
I've updated it now to not force an ordering on OutputType.